mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -588,22 +661,68 @@ static void cleanBranchOp(BranchOpInterface branchOp, RunLivenessAnalysis &la) {
       operandValues.push_back(successorOperands[operandIdx]);
     }
 
-    BitVector successorLiveOperands = markLives(operandValues, la);
-
     // Do (3)
-    for (int argIdx = successorLiveOperands.size() - 1; argIdx >= 0; --argIdx) {
-      if (!successorLiveOperands[argIdx]) {
-        if (successorBlock->getNumArguments() < successorOperands.size()) {
-          // if block was cleaned through a different code path
-          // we only need to remove operands from the invokation
-          successorOperands.erase(argIdx);
-          continue;
-        }
+    BitVector successorNonLive =
+        markLives(operandValues, nonLiveSet, la).flip();
+    collectNonLiveValues(nonLiveSet, successorBlock->getArguments(),
+                         successorNonLive);
+    cl.blocks.push_back({successorBlock, successorNonLive});
+    cl.successorOperands.push_back({branchOp, succIdx, successorNonLive});
+  }
+}
+
+static void cleanUpDeadVals(RDVFinalCleanupList &RDVFinalCleanupList) {
```

Comment:
Please add a comment. To me, something along these lines would be most helpful:
> Removes dead values collected in `RDVFinalCleanupList`. To be run once when _all_ dead values have been collected.

Feel free to re-use. 

I'd also be tempted to rename `&RDVFinalCleanupList` as `cleanupList` or even `list` (importantly, note camelCase rather than CamelCase). This hook is clear enough that using simpler names should be fine.

Reply:
done, thank you!